src/Api/Validator.php

```diff
@@ -91,10 +92,21 @@ private function check_structure(StructureShape $shape, $value)
     private function check_list(ListShape $shape, $value)
     {
         if (!is_array($value)) {
-            $this->addError('must be an array. Found ' . Aws\describe_type($value));
+            $this->addError('must be an array. Found '
+                . Aws\describe_type($value));
             return;
         }
 
+        if ($shape['min'] && count($value) < $shape['min']) {
```

Comment:
One minor optimization might be to store the count in a variable and compute it once